feat(choice): add hover style to choice card, fix active state

laurenhu opened the pull request:
<!--
  🤖 This repo uses Conventional Commits (conventionalcommits.org) to automate
  release notes and versioning. Title your PR using the following template:

  <type>(<scope>): <subject>

  Scope is optional. Indicate a breaking change by adding ! after the type/scope.

  Version influencing types:
  - fix: user-facing bug fix (patch version bump)
  - feat: user-facing feature (minor version bump)

  Other types:
  - docs: changes to the documentation
  - build: changes that affect the build system or external dependencies
  - test: adding missing tests, refactoring tests; no production code change
  - refactor: refactoring production code, eg. renaming a variable
  - style: formatting, missing semi colons, etc; no production code change
  - chore: updating grunt tasks etc; no production code change
  - revert: reverts a previous commit
  - perf: changes that improve performance
  - ci: changes to CI configuration files and scripts (eg. GitHub Actions)

  👍 Do examples:
  - feat(button): primary variant
  - fix(action-bar): inherit event-listeners

  👎 Don't examples:
  - feat(button): [ABC-123] primary variant

  Read CONTRIBUTING.md for more info.
-->

## Describe the problem this PR addresses
Choice cards do not have hover states
<!--
  🤐 If you are a Square employee, be mindful of any internal information
  you share in this public repository.
-->

## Describe the changes in this PR
Adds hover and a
<img width="619" alt="Screen Shot 2022-12-22 at 3 28 12 PM" src="https://user-images.githubusercontent.com/1486885/209220625-3f3ffae2-9489-48ab-b0a9-c315fd062566.png">
ctive states to ChoiceCard that match outline form components.

<!--
  📸 Inline screenshots to better communicate the changes
-->

## Other information
<!--
  🙆‍♂️ Provide further context that will help those out-of-the-loop
  to quickly understand the changes.
-->

pretzelhammer commented:
Is this PR outdated? I noticed on the latest-stable the selected color is the primary color:

![image](https://user-images.githubusercontent.com/7769424/211358058-f74a3b48-eb64-48a6-bad7-6c6a85c3d274.png)

But that on this branch it's just neutral-80:

![image](https://user-images.githubusercontent.com/7769424/211358270-efe73744-dddf-45ce-99da-b75ca4ca8404.png)

Also, I'm not sure how I feel about:
1. The hover state on ChoiceCards being identical to the selected state
2. ChoiceCards having a hover state but Choices not having a hover state

If we could solve 1 & 2 then that would be awesome.

laurenhu commented:
@pretzelhammer The decision to change the selected color of a choice card to always be neutral instead of the primary color like the choice options was based on design feedback our of a desire for its styling to more closely match the form elements. I've:
- Updated the select state on choice cards to be more distinct from the hover state
- Added a hover state for choice options.

pretzelhammer reviewed (approved):
the updates look great, thanks!
